### PR TITLE
[Compile Warnings As Errors] WordPress Module - Resolve Other Warnings

### LINKED ISSUE #17181: Compiler Warnings as Errors - WordPress Module
Parent #17173

This issue is about resolving/suppressing the compile warnings for the `WordPress` app module.

Number of Warnings: `621`
    - Main Source: `481`
    - Debug Source: `2`
    - WordPress Source: `5`
    - Jetpack Source: `4`
    - Test Source: `128`
    - UI Test Source: `1`

- [x] Resolve/Suppress all warnings.
    - [x] Main Source (`481`) @ParaskP7
        - [x] Resolve/Suppress `Coroutines` related warnings (`6`). @ParaskP7 #17210
            1. This declaration is experimental and its usage should be marked with '@kotlinx.coroutines.ExperimentalCoroutinesApi' or '@OptIn(kotlinx.coroutines.ExperimentalCoroutinesApi::class)' (`3`)
            2. 'XYZ' is deprecated. Implement CoroutineScope interface and cancel all child coroutines in xyz (`2`)
            3. @Synchronized annotation is not applicable to suspend functions and lambdas (`1`) (with [comment](https://github.com/wordpress-mobile/WordPress-Android/pull/10878#issuecomment-1258130096), cc @malinajirka)
        - [x] Resolve/Suppress `Deprecated` related warnings (`365`). @ParaskP7
            - [x] Specific
                1. `PreferenceManager` related warnings. @ParaskP7 #17215
                2. `LifecycleEvent` related warnings. @ParaskP7 #17223
            - [x] Generic 
                1. 'XYX' is deprecated. Deprecated in Java (`247` minus `Specific` above) @ParaskP7 #17246
                3. 'Xyz' is deprecated. Overrides deprecated member in 'xyz'. Deprecated in Java (`5`) @ParaskP7 #17287
                4. 'xyz' is deprecated. Use xyz instead. (`24`) @ParaskP7 #17287
                5. 'Xyz' is deprecated. This should be removed once we start receiving Xyz the source block from the backend (`2`) @ParaskP7 #17287
                6. 'Xyz' is deprecated. This class is being refactored, if you implement any change, please also update xyz (`87`) @ParaskP7 #17287
        - [x] Resolve/Suppress `Type/Cast/When` related warnings (`75`). @ParaskP7 #17291
            - Type (`44`)
                1. Unsafe use of a nullable receiver of type Xyz? (`4`)
                2. Type mismatch: inferred type is Xyz? but Xyz was expected (`13`)
                3. Elvis operator (?:) always returns the left operand of non-nullable type Xyz (`4`)
                4. The corresponding parameter in the supertype 'Xyz' is named 'xyz'. This may cause problems when calling this function with named arguments. (`4`)
                5. Unnecessary non-null assertion (!!) on a non-null receiver of type Xyz (`1`)
                7. Unnecessary safe call on a non-null receiver of type Xyz. This expression will have nullable type in future releases (`18`)
            - Cast (`5`)
                1. Unchecked cast: Xyz to Xyz (`4`)
                2. This cast can never succeed (`1`)
            - When (`26`)
                1. Non exhaustive 'when' statements on xyz will be prohibited in 1.7, add 'Xyz', 'Xyz' branch(es) or 'else' branch instead (`26`)
        - [x] Resolve/Suppress `Other` related warnings (`35`). @ParaskP7 #17293
            1. This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API. (`1`)
            2. Parameter 'xyz' is never used (`17`)
            3. Parameter 'xyz' is never used, could be renamed to _ (`9`)
            4. Name shadowed: xyz (`2`)
            5. Parcelize annotations from package 'kotlinx.android.parcel' are deprecated. Change package to 'kotlinx.parcelize' (`1`)
            6. Variable 'xyz' initializer is redundant (`1`)
            8. There is more than one label with such a name in this scope (`1`)
            9. Condition 'xyz' is always 'xyz' (`2`)
            10. Unreachable code (`1`)
    - [x] Debug/WordPress/Jetpack Source (`11`) @ParaskP7
        - [x] Resolve/Suppress `Deprecated` related warnings (`11`). @ParaskP7 #17287
            1. 'Xyz' is deprecated. Deprecated in Java (`11`)
    - [x] Test/UI Test Source (`129`) @ParaskP7
        - [x] Resolve/Suppress `Coroutines` related warnings (`11`). @ParaskP7 #17210
            1. This declaration is experimental and its usage should be marked with '@kotlinx.coroutines.ExperimentalCoroutinesApi' or '@OptIn(kotlinx.coroutines.ExperimentalCoroutinesApi::class)' (`8`)
            2. 'Xyz' is deprecated. Deprecated in the favour of 'xyz'. Please note that the provided replacement does not rethrow channel's close cause as 'poll' did, for the precise replacement please refer to the 'poll' documentation (`3`)
        - [x] Resolve/Suppress `Deprecated` related warnings (`80`). @ParaskP7 #17300
            1. 'XYX' is deprecated. Deprecated in Java (`58`)
            2. 'xyz' is deprecated. Use xyz instead. (`2`)
            3. 'Xyz' is deprecated. This class is being refactored, if you implement any change, please also update xyz (`20`)
        - [x] Resolve/Suppress `Type/Cast/When` related warnings (`15`). @ParaskP7 #17300
            - Type (`8`)
                1. Type mismatch: inferred type is Xyz? but Xyz was expected (`3`)
                2. Unnecessary non-null assertion (!!) on a non-null receiver of type Xyz (`4`)
                3. Unnecessary safe call on a non-null receiver of type Xyz. This expression will have nullable type in future releases (`1`)
            - Cast (`6`)
                1. Unchecked cast: Xyz to Xyz (`4`)
                2. This cast can never succeed (`2`)
            - When (`1`)
                1. Non exhaustive 'when' statements on xyz will be prohibited in 1.7, add 'Xyz', 'Xyz' branch(es) or 'else' branch instead (`1`)
        - [x] Resolve/Suppress `Other/Test/UI Test` related warnings (`23`). @ParaskP7 #17300
            - Other (`5`)
                1. Parameter 'xyz' is never used (`5`)
            - Test (`17`)
                1. Check for instance is always 'xyz' (`2`)
                2. Name contains characters which can cause problems on Windows: " (`15`)
            - UI Test (`1`)
                1. Name shadowed: fileName (`1`)
- [x] Enable all warnings as errors flag. @ParaskP7 #17318

### PR DESCRIPTION
**Pull request by ParaskP7**
Parent: #17173
Partially Closes: #17181

This PR resolves one part of all `Other` related warnings for the `WordPress` module, and in this part, (mainly) specific to the main source:

Main Source (`35`):
- `1` x `This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.`
- `17` x `Parameter 'xyz' is never used`
- `9` x `Parameter 'xyz' is never used, could be renamed to _`
- `2` x `Name shadowed: xyz`
- `1` x `Parcelize annotations from package 'kotlinx.android.parcel' are deprecated. Change package to 'kotlinx.parcelize'`
- `1` x `Variable 'xyz' initializer is redundant`
- `1` x `There is more than one label with such a name in this scope`
- `2` x `Condition 'xyz' is always 'xyz'`
- `1` x `Unreachable code`

Test Source (`5`)
- `3` x `Parameter 'xyz' is never used`
- `2` x `Variable 'xyz' is never used`

-----

Warnings Resolution List:

1. [Resolve global scope delicate coroutines api warnings.](https://github.com/wordpress-mobile/WordPress-Android/commit/d81c5ed16894b6013db5ec327e7acc603d26b25c)
2. [Resolve parameter is never and could be renamed to _ warnings.](https://github.com/wordpress-mobile/WordPress-Android/commit/528a9915f0bbc350593aabc7ff5729c4376397b2)
3. [Resolve variable is never used test warnings.](https://github.com/wordpress-mobile/WordPress-Android/commit/4a4f28014731434a65923608328a16e3528f2f0a)
4. [Resolve name shadowed warnings.](https://github.com/wordpress-mobile/WordPress-Android/commit/4eee7ad94afaebcd4638676a1d1555c2f5b37fd9)
5. [Resolve parcelize annotations deprecated package warning.](https://github.com/wordpress-mobile/WordPress-Android/commit/53b180c90ede495c316fded460a93cd1bd64fdd9)
6. [Resolve variable initializer is redundant warning.](https://github.com/wordpress-mobile/WordPress-Android/commit/bb6349c146c56d102767ae74c35f3bfb139f1f13)
7. [Resolve more than one label with such a name in scope warning.](https://github.com/wordpress-mobile/WordPress-Android/commit/32b8d595812f59f4ad34d7c2bec7712694e171a3)
8. [Resolve condition is always true warnings.](https://github.com/wordpress-mobile/WordPress-Android/commit/d6bd9512801cbc1a10a1a3ed60bd1075096ccd57)
9. [Resolve unreachable code warning.](https://github.com/wordpress-mobile/WordPress-Android/commit/306046922c1631fe3b97a8ea495147a98f73ee6e)

Warnings Suppression List:

1. [Suppress parameter is never used warnings.](https://github.com/wordpress-mobile/WordPress-Android/commit/ac8fbda82eeb7df50db619978baec1cb355d4fa9)

Refactor List:

1. [Use suppress in favor of suppress warnings for kotlin classes.](https://github.com/wordpress-mobile/WordPress-Android/commit/66688f456786581ee971ffcea8a5e6cca92f77fe)

-----

PS: @zwarm I added you as the main reviewer, that is, in addition to @wordpress-mobile/apps-infrastructure team itself, but randomly, since I want someone from the WordPress Android team to primarily sign-off on that change. 🥇

FYI: I am going to randomly add more of you in those PRs that will follow, just so you become more aware of this change and how close we are on enabling allWarningsAsErrors by default everywhere. 🎉

-----

To test:

- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough.
- However, if you want to be thorough about reviewing this change, you could quickly smoke test both, the `WordPress` and `Jetpack` apps, and see if they are both working as expected.

-----

## Regression Notes

1. Potential unintended areas of impact

N/A

10. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

11. What automated tests I added (or what prevented me from doing so)

N/A

-----

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/utils/StatsUtils.kt [hunk 1]
```diff
@@ -173,7 +173,7 @@ class StatsUtils
         entries: List<Line>
     ): List<String> {
         val contentDescriptions = mutableListOf<String>()
-        entries.forEachIndexed { index, bar ->
+        entries.forEachIndexed { _, bar ->
```
Comment (zwarm) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/utils/StatsUtils.kt:176]: @ParaskP7 - all is good with the change you made ... just questioning why the use of `forEachIndexed` without actually using the index. 🤷 What am I missing? And I don't expect a change for this, it's not part of the PR.

**Review by zwarm — APPROVED** [review]
@ParaskP7 - I have taken the apps out for a spin. Nothing unusual, so 👍 . Thanks for including me a reviewer, it's been nice seeing the changes and helps me in the future. I left one question, non-PR related, so I'll be merging this PR.

🚢

**Comment by ParaskP7** [comment]
And again, thank you so much for reviewing, testing and merging this PR @zwarm, you are awesome! 🙇 ❤️ 🚀

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/utils/StatsUtils.kt (hunk 1 above)
Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/utils/StatsUtils.kt:176]: Ha, yea, you are right @zwarm, this could have further improved to become a plain `entries.forEach { bar -> ... }`, the indexed version is not needed as the `index` ends up becoming unused. I am seeing this `entries.forEachIndexed { index, bar -> ... }` being actually used within `getBarChartEntryContentDescriptions(...)` above, and as such, I think this was just a copy-pasting overside at that point to create the `getLineChartEntryContentDescriptions(...)` below, which this time didn't have any need for that.

Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/utils/StatsUtils.kt:176]: PS: I am going to fix that on my next PR. It would be just another refactoring commit on top of all the others commits! 🎉 Thanks for spotting it! 🙇 ❤️ 🌟
